### ISSUE TITLE
analysz VOLATILITY IX

### ISSUE THREAD

**Issue by aymenhma99** [aymenhma99/volatility-75#1]
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Load your dataset
# data = pd.read_csv('path_to_your_data.csv')

# For demonstration, let's create a dummy dataset
np.random.seed(0)
dates = pd.date_range('20210101', periods=100)
data = pd.DataFrame(np.random.randn(100, 1), index=dates, columns=['Volatility'])

# Feature Engineering: Create lagged returns as features
data['Lagged_1'] = data['Volatility'].shift(1)
data['Lagged_2'] = data['Volatility'].shift(2)
data.dropna(inplace=True)

# Define your target variable and features
X = data[['Lagged_1', 'Lagged_2']]
y = data['Volatility']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Initialize and train the model
model = LinearRegression()
model.fit(X_train, y_train)

# Make predictions
predictions = model.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, predictions)
print(f'Mean Squared Error: {mse}')

# You can now use the model to predict future volatility
